### DISCUSSION TITLE
Config and general questions from a newbie

### DISCUSSION THREAD

**Discussion by Peronia** [qdm12/ddns-updater#648] [Q&A] [answered]
Hi,
first thanks for this project, is exactly what I'm looking for. I looked at your config section and got 2 questions:

1. What exactly is `host` (or `owner` later)? I can use a subdomain (or more), @ and *, got it. But what is this field?
2. The config have a field `domain`, clear. But what if I want 2 domanis at the same provider to get updated, like example.com and www.example.com or example1.com and example2.com? Every domain needs a new entry?

Not really config questions:

- The standalone binary has an inbuild webserver, so I don't need one, true?
- A new DNS provider or domain registrar must have an API, true?
- Call me stupid but I wonder: When I use this ddns updater, I get rid of the URL which my router updates each time the ip address changes and the CNAME entry at my domain for that (and of course other dns updates at other registrars, that got handled over scripts etc.)? I mean it sounds so and technically I can imagine this but this would be very nice.
- Can I avoid that a cached entry from e.g. my pihole/other dns servers get used?

**Comment by qdm12** [comment]
Hi there!

1. As you've seen in https://github.com/qdm12/ddns-updater/issues/625 the technical term is `owner`, although on some registrars they do use `host`. It's called owner because it's the name of the owner for a particular RR (resource record, for example A type RR) for a domain name. For example owner `a` owns the A type RR for domain `example.com`, meaning querying A a.example.com will return the value associated with the A type RR... owned by `a`. It's probably some historical reason behind this, but if you find a better name (not `subdomain`), feel free to comment! 😉 
2. You can comma separate hosts (owners) with `a,b` for the same domain, you need separate entries for each domain name. But I would have to do it again, I would had refrained from having multiple hosts as comma separated, it just makes it more complicated than just copy pasting and having the same structure. So I am not going to do this for domain unless there is a very compelling reason to do so.

> The standalone binary has an inbuild webserver, so I don't need one, true?

It's just a static web UI to give you information on your domains/public IP address/errors, you definitely need another server (or proxy like Caddy) to serve anything useful to the outside world.

> A new DNS provider or domain registrar must have an API, true?

Yes, or an update URL. Check the list of issues, there are already a lot requested waiting in queue, maybe the one you want is there: https://github.com/qdm12/ddns-updater/issues Feel free to upvote 👍 one and I'll notice it eventually. Right now the priority is to add features and not more providers (unless contributed by people).

> Call me stupid but I wonder: When I use this ddns updater, I get rid of the URL which my router updates each time the ip address changes and the CNAME entry at my domain for that (and of course other dns updates at other registrars, that got handled over scripts etc.)? I mean it sounds so and technically I can imagine this but this would be very nice.

I'm not sure I fully understand, especially about the CNAME entry 🤔 If you mean this does the job all at the same time for N providers, yes it does 😉 You can also use the custom provider (rather new) to specify your own update URL.

> Can I avoid that a cached entry from e.g. my pihole/other dns servers get used?

You can use `RESOLVER_ADDRESS=1.1.1.1` or `--resolver-address 1.1.1.1` to use Cloudflare DNS servers instead of Pihole for the program (it may use a bit of caching from Cloudflare, but usually they somehow find out when your records get updated quite quickly).

**Comment by Peronia** [comment]
Thanks for your answers. Most of my questions are solved, but some remain:

- Uff, host/owner is hard to understand, is the owner a person or the registrar of the domain? Neverless, when I "own" example.com and put @ at the `host` value, then only the entry (A/AAAA) itself are updated (specified in `domain`)? When I put `host` to * every record (A/AAAA/MX/TXT/...) for this domain get updated? And lastly, when I put hello.example.com into `host`, only this entry get a new IP? When all of this are correct I got it.
- Sure, the binary don't offer any reverse proxy or like that, but I can visit the URL and see the static page (which hopefully get updated when the IP changes and so on)?
- My actual setup:
  - The router have a classic dd client (checks the DNS record of a secified subdomain and when this don't match the actual IP, it would get an update)
  - Then I have 2 scripts (one for each registrar) that checks the root domain (example.com, without a subdomain) against the subdomain that is set by the router dd client. If this doesn't fit, change it. It seems a bad idea to use a cname for the root domain, this is the reason for me to buil it this way: 
![grafik](https://github.com/qdm12/ddns-updater/assets/25049991/cd1736aa-8b95-45c0-98ba-1d63961682eb)

**Comment by qdm12** [comment, marked as answer]
> Uff, host/owner is hard to understand

RR type is A, AAAA, MX etc.
RR owner is @, *, abc, etc. and is the part prefixing the fully qualified domain name (FQDN) so `ab` is the host for `ab.domain.com` and `domain.com` is the domain

> which hopefully get updated when the IP changes and so on

Yes, although it needs a page reload

> My actual setup

Ah yes that can be replaced with this program indeed.

**Comment by Peronia** [comment]
Thanks, now it is clear to me what host/owner is. Maybe it should descripted with an example in the readme, like you did it in the last comment.
